File: src/scala/BubbleSort.scala

def bubble_sort(data: List[Int]): List[Int] = {
  var sortedData: List[Int] = data
  var swapped: Boolean = true
  while (swapped == true) {
    swapped = false
    for (i <- 0 to sortedData.length - 2) {
      if (sortedData(i) > sortedData(i+1)) {
        sortedData = sortedData.updated(i, sortedData(i+1)).updated(i+1, sortedData(i))
        swapped = true
      }
    }
  }
  sortedData
}

object Main extends App {
  var data: List[Int] = List(9, 2, 8, 1, 5, 7, 3, 6, 4)
  println(data)
  var sortedData: List[Int] = bubble_sort(data)
  println(sortedData)
}

File: src/javascript/BubbleSort.js
function bubbleSort(unsortedArray, start, end) {
    let swapped;

    do {
        swapped = false;
        for (var i = start; i < end - 1; i++) {
            if (unsortedArray[i] > unsortedArray[i + 1]) {
                [unsortedArray[i], unsortedArray[i + 1]] = [unsortedArray[i + 1], unsortedArray[i]];
                swapped = true;
            }
        }
    } while (swapped);

    return unsortedArray;
}

var unsortedArray = [54, 42, 11, 33, 24, 99, 77, 80];
let sortedArrayViaBubbleSort = bubbleSort(unsortedArray, 0, unsortedArray.length);

console.log(sortedArrayViaBubbleSort);